File: day1/p2/solution.rb
def captcha(nums)
  nums.each_with_index.reduce(0) do |sum, (num, i)|
    next_index = ((nums.length / 2) + i) % nums.length
    sum + (num == nums[next_index] ? nums[next_index] : 0)
  end
end

nums = File.read('input.txt').split('').map(&:to_i)
p captcha(nums)

File: day1/p2/solution.exs
{:ok, input} = File.read("input.txt")

nums = input
|> String.split("", trim: true)
|> Enum.map(fn(num) -> String.to_integer(num) end)

{result, _} = nums
|> Enum.reduce({0, 0}, fn(num, {agg, i}) ->
  full_index = round((length(nums) / 2) + i)
  next_num = Enum.at(nums, rem(full_index, length(nums)))
  to_sum = if (num == next_num), do: next_num, else: 0
  {agg+to_sum, i+1}
end)

IO.inspect(result)
